src/teststat/BinnedNLLH.cpp

```diff
@@ -62,25 +62,49 @@ BinnedNLLH::BinData(){
 }
 
 void
-BinnedNLLH::SetPdfManager(const BinnedEDManager& man_){
-    fPdfManager = man_;
+BinnedNLLH::AddDist(const std::vector<BinnedED>& pdfs, const std::vector<std::vector<std::string> >& sys_){
+    if (pdfs.size() != sys_.size())
+       throw DimensionError(Formatter()<<"BinnedNLLH:: #sys_ != #group_");
+    for (int i = 0; i < pdfs.size(); ++i)
+        AddDist( pdfs.at(i), sys_.at(i) );
 }
 
 void
-BinnedNLLH::SetSystematicManager(const SystematicManager& man_){
-    fSystematicManager = man_;
+BinnedNLLH::AddDist(const BinnedED& pdf_, const std::vector<std::string>& syss_){
+    fPdfManager.AddPdf(pdf_);
+    fSystematicManager.AddDist(pdf_,syss_);
 }
 
 void
-BinnedNLLH::AddPdf(const BinnedED& pdf_){
+BinnedNLLH::AddDist(const BinnedED& pdf_){
     fPdfManager.AddPdf(pdf_);
+    fSystematicManager.AddDist(pdf_,"");
```

Comment:
Not sure why not just add this line to AddPdf(). Similarly, why not call AddDist(const BinnedED& pdf_, const std::vector<std::string>& syss_)  above (line 73) also AddPdf(...) etc.